valispace/__init__.py

```diff
@@ -62,18 +62,31 @@ def __init__(self, url=None, username=None, password=None):
 		except:
 			print "VALISPACE-ERROR: Invalid credentials or url"
 
-	def all_valis(self):
+	def all_data(self, type=None):
 		"""
-		Returns a dict of all Valis with their properties
+		Returns a dict of all component/vali/textvali/tags with their properties
 		"""
+		# Check if no argument was passed
+		if type is None:
+			print "VALISPACE-ERROR: Type argument expected (component/vali/textvali/tags)"
+			return
+
+		# URL
+		if type is 'component':
```

Comment:
Maybe change to switch/case structure?

Reply:
I don't think Python have switch/case. I can use dictionary mapping but I don't think that is nicer than if/elif [https://www.pydanny.com/why-doesnt-python-have-switch-case.html](https://www.pydanny.com/why-doesnt-python-have-switch-case.html)

Reply:
OK yes I was confused with JavaScript :)